api/classes/class-lightweight-api.php

```diff
@@ -195,7 +245,10 @@ public function get_campaign_data( $client_id, $campaign_id ) {
 	 * @param string $campaign_data Prompt data.
 	 */
 	public function save_campaign_data( $client_id, $campaign_id, $campaign_data ) {
-		return $this->set_transient( $this->get_transient_name( $client_id, $campaign_id ), $campaign_data );
+		$client_data                            = $this->get_client_data( $client_id, true );
+		$client_data['prompts'][ $campaign_id ] = $campaign_data;
+
+		return $this->set_transient( $client_id, $client_data );
```

Comment:
Tried this in d9bf5fd. Note that in the `save_client_data` function, I had to replace `array_merge_recursive` with `array_replace_recursive` because the former was giving me a result that merged single array values into arrays, e.g. 

```
'count' => [ 0, 1, 2 ]
```

[`array_replace_recursive`](https://www.php.net/manual/en/function.array-replace-recursive.php) seems closer to the behavior of `array_merge` anyway, right?

Reply:
0b4fc2b fixes the failing tests—`array_merge_recursive` is still needed for the non-prompt client data, since we don't actually want to replace `posts_read` but append to it, for example.